src/silo/prepare_dataset.cpp

```diff
@@ -239,6 +239,19 @@ void silo::partitionData(
       );
    }
 
+   for (const auto& [aa_name, reference_genome] : reference_genomes.aa_sequences) {
+      std::filesystem::path sequence_filename = input_folder;
+      sequence_filename += "gene_" + aa_name + ".fasta";
+      FastaReader sequence_input(sequence_filename);
+
+      std::filesystem::path aa_folder = output_folder;
+      aa_folder += "gene_" + aa_name + std::filesystem::path::preferred_separator;
+
+      create_directory(aa_folder);
+
+      partitionSequenceFile(sequence_input, aa_folder, chunk_names, key_to_chunk, reference_genome);
+   }
+
```

Comment:
This is very similar to the previous for loop. How easy would it be to extract a function?